### PR TITLE
[OpenVINO] Fix backend applying dropout during predict

### PR DESCRIPTION
**Pull request by goyaladitya05**
## Description
`_get_compiled_model` was calling self(self.struct_params) without training=False when tracing the Keras model into an OpenVINO graph. This caused training-mode ops (e.g. Dropout) to be into the compiled model, so dropout was incorrectly applied during inference.

All other backends (TF, Torch, JAX, NumPy) guard their predict/test steps with _call_has_training_arg and pass training=False. This PR mirrors that pattern in the OpenVINO _get_compiled_model path.

## Contributor Agreement
**Please check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

> **Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/backend/openvino/trainer.py [hunk 1]
```diff
@@ -143,7 +143,10 @@ def _get_compiled_model(self, data):
             data, dynamic_batch_size=True
         )
         # construct OpenVINO graph during calling Keras Model
-        self.struct_outputs = self(self.struct_params)
+        if self._call_has_training_arg:
```
[the remark this answers is not in the document]
Comment (goyaladitya05) [reply keras/src/backend/openvino/trainer.py:146]: the other backends also access it directly.

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
